addons/nuxeo-retention/nuxeo-retention-core/src/main/java/org/nuxeo/retention/service/RetentionManagerImpl.java

```diff
@@ -0,0 +1,363 @@
+            return;
+        }
+        if (events.contains(startingPointEvent)) {
+            ELActionContext actionContext = initActionContext(record.getDocument(), session);
+            String expression = rule.getStartingPointExpression();
+            boolean startNow = evaluateConditionExpression(actionContext, expression);
+            if (startNow) {
+                session.setRetainUntil(record.getDocument().getRef(), rule.getRetainUntilDateFromNow(), null);
+                log.debug("Evaluating event-based rules: expression {} matched on event {}", expression,
+                        startingPointEvent);
+            } else {
+                log.debug("Evaluating event-based rules: expression {} did not match on event {}", expression,
+                        startingPointEvent);
+            }
+        }
+    }
+
+    @Override
+    public void proceedRetentionExpired(Record record, CoreSession session) {
+        executeRuleEndActions(record, session);
+    }
+
+    protected volatile List<String> acceptedEvents; // NOSONAR double-checked locking
+
+    @Override
+    public List<String> getAcceptedEvents() {
+        if (acceptedEvents == null) {
+            synchronized (this) {
+                if (acceptedEvents == null) {
+                    DirectoryService directoryService = Framework.getService(DirectoryService.class);
+                    Directory dir = directoryService.getDirectory(RetentionConstants.EVENTS_DIRECTORY_NAME);
+                    try (Session session = dir.getSession()) {
+                        Map<String, Serializable> filter = new HashMap<>();
+                        filter.put(RetentionConstants.OBSOLETE_FIELD_ID, Long.valueOf(0));
+                        List<String> evts = session.getProjection(filter, session.getIdField());
+                        if (evts.isEmpty()) {
+                            log.trace("Empty accepted events, not catching anything");
+                            return evts;
+                        }
+                        acceptedEvents = evts;
+                        log.debug("Accepted events {}", acceptedEvents::toString);
+                    }
+                }
+
+            }
+        }
+        return acceptedEvents;
+    }
+
+    @Override
+    public void invalidate() {
+        synchronized (this) {
+            acceptedEvents = null;
+        }
+    }
+
+    @Override
+    public void start(ComponentContext context) {
+        Framework.doPrivileged(() -> {
+            UserManager userManager = Framework.getService(UserManager.class);
+            if (userManager.getGroup(RetentionConstants.RECORD_MANAGER_GROUP_NAME) == null) {
+                DocumentModel groupModel = userManager.getBareGroupModel();
+                String groupSchemaName = userManager.getGroupSchemaName();
+                groupModel = userManager.getBareGroupModel();
+                groupModel.setPropertyValue(userManager.getGroupIdField(),
```

Comment:
redundant declaration here: `DocumentModel groupModel = userManager.getBareGroupModel();`   and `groupModel = userManager.getBareGroupModel();`